packages/taquito-beacon-wallet/src/taquito-beacon-wallet.ts

```diff
@@ -186,34 +186,35 @@ export class BeaconWallet implements WalletProvider {
     await this.client.setActiveAccount();
   }
 
-  async sign(signingRequest: { payload: string; watermark?: Uint8Array }) {
-    let bb = hex2buf(signingRequest.payload);
-    if (typeof signingRequest.watermark !== 'undefined') {
-      bb = mergebuf(signingRequest.watermark, bb);
+  async sign({ payload, watermark }: { payload: string; watermark?: Uint8Array }) {
+    let bb = hex2buf(payload);
+    if (typeof watermark !== 'undefined') {
+      bb = mergebuf(watermark, bb);
     }
     const watermarkedBytes = buf2hex(toBuffer(bb));
-
-    let signingType: SigningType;
-    switch (signingRequest.watermark) {
-      case new Uint8Array([5]):
-        signingType = SigningType.MICHELINE;
-        break;
-      case new Uint8Array([3]):
-        signingType = SigningType.OPERATION;
-        break;
-      case undefined:
-        signingType = SigningType.RAW;
-        break;
-      default:
-        throw new Error(`Invalid watermark ${JSON.stringify(signingRequest.watermark)}`);
-    }
+    const signingType = this.getSigningType(watermark);
     const { signature } = await this.client.requestSignPayload({
       payload: watermarkedBytes,
       signingType,
     });
     return signature;
   }
 
+  private getSigningType(watermark?: Uint8Array) {
+    if (!watermark || watermark.length === 0) {
+      return SigningType.RAW;
```

Comment:
Just a suggestion, beacon has this warning that signing raw can get different sig from wallet implementations that it'll be difficult to verify the sig, at this point we might not want to have signing raw available to avoid user not happy that it's a bug we introduce
<img width="869" alt="image" src="https://github.com/ecadlabs/taquito/assets/106410553/4106a078-feb9-4239-9866-f1c3c23d2bac">